### PR TITLE
Person URLs in Funnels

### PR DESCRIPTION
**Pull request by pauldambra**
## Changes

There were two things combining so that Funnel person URLs could sometimes end up starting `https://api/etc` instead of `https://api.posthog.com`

Most paths through creating a funnel don't add a base_uri. Which means the funnel defaults to an absolute url with no protocol e.g. `/api/etc`

One path through added the current base_uri from the request. For cloud `https://api.posthog.com`

That same path through for cached insights has no request and so added no base uri

-----

The front end had code in `api.get` which ensured a leading and trailing slash for URLs with no protocol.

It didn't check of the URL already had a leading slash and so `/api/etc` was converted to `//api/etc` which the browser renders as `https://api/etc`

-----

This change does two things

1) makes sure the front end only adds a leading slash when it isn't already present.
2) doesn't ever add a base_uri in the backend since everything works fine without it

## How did you test this code?

running locally and adding unit tests

### CODE REVIEW

File: frontend/src/lib/api.test.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            fakeFetch = jest.fn()
+            window.fetch = fakeFetch
+        })
+
+        it('adds a leading slash to relative URLs', () => {
+            api.get('relative/url')
+            expect(fakeFetch.mock.calls[0][0]).toEqual('/relative/url/')
+        })
+
+        it('does not add leading slash when absolute url with no http', () => {
```
Comment (hazzadous) [inline frontend/src/lib/api.test.ts:17]: 💯

File: ee/clickhouse/views/insights.py [hunk 2]
```diff
@@ -107,8 +107,7 @@ def calculate_funnel(self, request: Request) -> Dict[str, Any]:
                 ).run()
             }
         else:
-            base_uri = request.build_absolute_uri("/")
-            return {"result": funnel_order_class(team=team, filter=filter, base_uri=base_uri).run()}
+            return {"result": funnel_order_class(team=team, filter=filter).run()}
```
Comment (hazzadous) [inline ee/clickhouse/views/insights.py:110]: I guess this is just to be able to debug the issue? Would you suggest reverting this change before merge or leaving in?

File: frontend/src/lib/api.ts [hunk 3]
```diff
@@ -236,10 +236,15 @@ const api = {
     },
 
     async get(url: string, signal?: AbortSignal): Promise<any> {
+        console.log(url, 'starting')
```
Comment (hazzadous) [inline frontend/src/lib/api.ts:239]: Remove me

File: frontend/src/lib/api.ts [hunk 4]
```diff
@@ -236,10 +236,15 @@ const api = {
...
         if (url.indexOf('http') !== 0) {
-            url = '/' + url + (url.indexOf('?') === -1 && url[url.length - 1] !== '/' ? '/' : '')
-        }
+            if (!url.startsWith('/')) {
```
Comment (hazzadous) [inline frontend/src/lib/api.ts:241]: Perhaps a bit of scope creep, but I guess this is also an issue with the other verbs.

Comment (hazzadous) [reply frontend/src/lib/api.ts:241]: Although non-blocking, if this solves the immediate issue and updating elsewhere is a more risky task.

File: frontend/src/lib/api.ts [hunk 5]
```diff
@@ -236,10 +236,15 @@ const api = {
...
+                url = '/' + url
+            }
 
+            url = url + (url.indexOf('?') === -1 && url[url.length - 1] !== '/' ? '/' : '')
+        }
+        console.log(url, 'transformed')
```
Comment (hazzadous) [inline frontend/src/lib/api.ts:247]: Remove me

File: ee/clickhouse/views/insights.py (hunk 2 above)
Comment (pauldambra) [reply ee/clickhouse/views/insights.py:110]: I was thinking that since it works without the base_uri being set. It's nicer to have all of the ways that we create `funnel_order_class` be more similar

File: frontend/src/lib/api.ts (hunk 3 above)
Comment (pauldambra) [reply frontend/src/lib/api.ts:239]: dangnammit

File: frontend/src/lib/api.ts (hunk 4 above)
Comment (pauldambra) [reply frontend/src/lib/api.ts:241]: Not scope creep, good thinking!

File: ee/clickhouse/views/insights.py (hunk 2 above)
Comment (pauldambra) [reply ee/clickhouse/views/insights.py:110]: (open to being challenged on that!)

**Review by hazzadous — APPROVED** [review]
